components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -0,0 +1,956 @@
+			map.resulttype = 'np'
+		elseif map.winner == '1' then
+			map.scores = { 1, 0 }
+		elseif map.winner == '2' then
+			map.scores = { 0, 1 }
+		elseif map.winner == '0' or map.winner == 'draw' then
+			map.scores = { 0.5, 0.5 }
+			map.resulttype = 'draw'
+		end
+	end
+	return map
+end
+
+function WarcraftMatchGroupInput.ProcessPlayerMapData(map, match, OppNumber)
+	local participants = {}
+	local map_mode = ''
+	local raceOP = {}
+	local PL = {}
+	local heroes = {}
+
+	for i = 1, OppNumber do
+		local number = 0
+		if match['opponent' .. i]['type'] == 'team' then
+			local players = match['opponent' .. i].match2players
+			if players == {} then
+				break
+			end
+
+			local tbds = 0
+
+			local PlayerData = {}
+			for j = 1, 4 do
+				if not Logic.isEmpty(map['t' .. i .. 'p' .. j]) then
+					if map['t' .. i .. 'p' .. j] ~= 'TBD' and map['t' .. i .. 'p' .. j] ~= 'TBA' then
+						map['t' .. i .. 'p' .. j] = mw.ext.TeamLiquidIntegration.resolve_redirect(map['t' .. i .. 'p' .. j])
+
+						PlayerData[map['t' .. i .. 'p' .. j]] = {
+							faction = FACTIONS[string.lower((map['t' .. i .. 'p' .. j .. 'race'] or '') ~= '' and
+								map['t' .. i .. 'p' .. j .. 'race'] or 'u')] or 'u',
+							position = j
+						}
+					else
+						tbds = tbds + 1
+					end
+				else
+					break
+				end
+			end
+
+			for key, item in pairs(players) do
+				if item and PlayerData[item.name] then
+					number = number + 1
+					local faction = (PlayerData[item.name].faction ~= 'u') and PlayerData[item.name].faction or
+						item.extradata.faction or 'u'
+					raceOP[i] = faction
+					PL[i] = item.name
+					participants[i .. '_' .. key] = {
+						faction = faction,
+						player = item.name,
+						position = PlayerData[item.name].position,
+						flag = cleanFlag(item.flag),
+					}
+				end
+			end
+
+			local num = #players
+
+			for r = 1, tbds do
+				number = number + 1
+				participants[i .. '_' .. (num + r)] = {
+						faction = 'u',
+						player = 'TBD'
+					}
+			end
+			
+			if number == 2 and map['opponent' .. i .. 'duoSpecial'] == 'true' then
+				number = '2S'
+			elseif number == 4 and map['opponent' .. i .. 'quadSpecial'] == 'true' then
+				number = '4S'
+			end
+		elseif match['opponent' .. i]['type'] == 'literal' then
+			number = 'Literal'
+		else
+			number = tonumber(MODES2[match['opponent' .. i]['type']])
+			local players = match['opponent' .. i].match2players
+			if players == {} then
+				break
+			else
+				local faction = string.lower((map['t' .. i .. 'p1race'] or '') ~= '' and map['t' .. i .. 'p1race'] or
+					(map['race' .. i] or '') ~= '' and map['race' .. i] or 'u')
+				local hero = string.lower(map['t' .. i .. 'plheroes'] or '') ~= '' and map['t' .. i .. 'plheroes']
+				participants[i .. '_1'] = {
+					faction = FACTIONS[faction] or players[1].extradata.faction or 'u',
+					player = players[1].name,
+					hero = HEROES[hero]
+				}
+				raceOP[i] = participants[i .. '_1'].faction
+				heroes[i] = hero[i].hero
```

Comment:
will error
hero is a plain string or `false` here, so you try to index a string and try to index that again